ranger-core/src/main/java/io/appform/ranger/core/serviceprovider/ServiceProvider.java

```diff
@@ -69,10 +76,30 @@ private void handleHealthUpdate(HealthcheckResult result) {
             log.debug("No update to health state of node. Skipping data source update.");
             return;
         }
+        setStartupTimeIfAbsent(result);
+        if (null != weightSupplier && weightSupplier.isEnabled() && isBetweenZeroAndOne(
+                weightSupplier.getSupplier().get())) {
+            serviceNode.setWeight(weightSupplier.getSupplier().get());
+        }
         serviceNode.setHealthcheckStatus(result.getStatus());
         serviceNode.setLastUpdatedTimeStamp(result.getUpdatedTime());
         dataSink.updateState(serializer, serviceNode);
         log.debug("Updated node with health check result: {}", result);
     }
 
+    private void setStartupTimeIfAbsent(final HealthcheckResult result) {
+        if (healthy == result.getStatus()) {
+            if (serviceNode.getNodeStartupTimeInMs() == 0) {
+                serviceNode.setNodeStartupTimeInMs(Instant.now().toEpochMilli());
```

Comment:
In some places you have used System.current.. and others instant.now .. fix either one